main/src/library/Console.flix

```diff
@@ -50,6 +50,24 @@ mod Console {
 
     import java.lang.System
 
+    ///
+    /// Reads a 32-bit integer from the console.
+    ///
+    pub def readInt32(): Option[Int32] \ Console =
+        Console.readln() |> Int32.fromString
+
+    ///
+    /// Reads a 32-bit integer from the console with a prompt.
+    ///
+    /// Repeatedly prints `msg` and reads input until a valid integer in the given range `[start; end)` is entered.
+    ///
+    pub def promptInt32(start: {start = Int32}, end: {end = Int32}, msg: a): Int32 \ Console with ToString[a] =
+        Console.println("${msg}");
+        match readInt32() {
+            case Some(i) if (start#start <= i and i < end#end) => i
+            case _                                             => promptInt32(start, end, msg)
+        }
+
```

Comment:
I think the generalized version is also pretty nice. This also to do allow `quit` to exit the loop fx
```flix
///
/// Reads a parsed input from the console with a prompt.
///
/// Repeatedly prints `msg` and reads input until a valid input is entered.
///
pub def prompt(parse: String -> Option[t] \ ef, msg: a): t \ Console + ef with ToString[a] =
    Console.println("${msg}");
    match Console.readln() |> parse {
        case Some(v) => v
        case _       => prompt(parse, msg)
    }
```
```flix
prompt(Int32.fromString >> (i -> start <= i and i < end))
```